filebeat/harvester/log.go

```diff
@@ -180,9 +181,13 @@ func (h *Harvester) sendEvent(event *input.Event) bool {
 // is started. As soon as the output becomes available again, the finished state is written
 // and processing can continue.
 func (h *Harvester) sendStateUpdate() {
+
 	logp.Debug("harvester", "Update state: %s, offset: %v", h.state.Source, h.state.Offset)
 	event := input.NewEvent(h.state)
-	h.outlet.OnEvent(event)
+	h.updateState(event.State)
```

Comment:
the `Harvest` go-routine should do the final update to `h.updateState` in a defer-statement. To ensure most recent state is available in prospector state map and mark harvester/state as released even in case of harvester panic'ing.